### PR TITLE
Removed deprecations from v5

### PR DESCRIPTION
**Pull request by ZurabDev**
BC

Removed module PrizmCounterModule use standalone 
Removed module PrizmButtonModule use standalone 

Removed deprecated module PrizmFlagIconsModule

Removed deprecated CallFuncPipe (use PrizmCallFuncPipe)
Removed deprecated module PrizmCallFuncModule

Removed deprecated module PrizmCallFuncModule
Removed deprecated module PrizmPluckModule
Removed deprecated module PrizmToObservableModule
Removed deprecated module PrizmToTypeModule

Removed deprecated module PrizmOverlayModule

Removed deprecated export ToTypePipe

Removed deprecated module PrizmLetModule
Removed deprecated function prizmCreateTimeMask, prizmCreateTimePartMask
Removed deprecated function prizmExtractI18n

Removed tokens PRIZM_DATE_RANGE_FILLER, PRIZM_DATE_FILLER

Removed functions prizmExtractI18n

Removed enums PrizmBaseColor, PrizmSupportColor

Removed deprecated module PrizmSkeletonModule
Removed deprecated module PrizmWrapperModule
Removed deprecated module PrizmScrollIntoViewModule
Removed deprecated module PrizmRepeatTimesModule
Removed deprecated module PrizmPreventDefaultModule
Removed deprecated module PrizmPressedModule

Removed deprecated module PrizmOverscrollModule
Removed deprecated module PrizmInputNativeValueModule
Removed deprecated module PrizmMutationObserveModule
Removed deprecated module PrizmLifecycleModule
Removed deprecated module PrizmHoveredModule
Removed module PrizmFocusedModule
Removed module PrizmFocusableModule
Removed module PrizmFocusTrapModule

Removed deprecated module PrizmDropdownZoneModule
Removed deprecated module PrizmDroppableModule
Removed deprecated module PrizmElementReadyModule
Removed deprecated module PrizmDropdownControllerModule

Removed module PrizmCheckedModule use standalone PrizmCheckedDirective
Removed module PrizmAutoResizeModule use standalone PrizmAutoResizeDirective
Removed module PrizmAutoFocusModule use standalone PrizmAutoFocusDirective
Removed module PrizmStopPropagationModule use standalone PrizmStopPropagationDirective
Removed module PrizmZoneEventModule use standalone PrizmZoneEventDirective
Removed module PrizmCalendarSheetModule use standalone PrizmCalendarSheetPipe
Removed module PrizmMapperModule use standalone PrizmMapperPipe
Removed module PrizmMonthModule use standalone PrizmMonthPipe

Removed deprecated module PrizmIconsSvgModule
Removed deprecated module PrizmShadowModule

Removed deprecated module PrizmWidgetModule
Removed deprecated module PrizmTreeButtonModule
Removed deprecated module PrizmToggleModule
Removed deprecated module PrizmSwitcherModule
Removed deprecated module PrizmSpinnerModule

Removed module PrizmChartsAreaModule use standalone PrizmChartsAreaComponent
Removed module PrizmChartsBarModule use standalone PrizmChartsBarComponent
Removed module PrizmChartsColumnModule use standalone PrizmChartsColumnComponent
Removed module PrizmChartsGaugeModule use standalone PrizmChartsGaugeComponent
Removed module PrizmChartsLineModule use standalone PrizmChartsLineComponent
Removed module PrizmChartsPieModule use standalone PrizmChartsPieComponent
Removed module PrizmChartsRadarModule use standalone PrizmChartsRadarComponent
Removed module PrizmChartsRadialBarModule use standalone PrizmChartsRadialBarComponent
Removed module PrizmChartsScatterModule use standalone PrizmChartsScatterComponent
Removed module PrizmChartsTreemapModule use standalone PrizmChartsTreemapComponent
Removed module PrizmChartsWaterfallModule use standalone PrizmChartsWaterfallComponent

Removed deprecated module PrizmRadioButtonModule

Removed module PrizmPanelModule use standalone PrizmPanelComponent
Renamed PanelComponent to PrizmPanelComponent

Removed deprecated module PrizmPaginatorModule
Removed deprecated module PrizmLoaderModule
Removed deprecated module PrizmLinkModule
Removed deprecated module PrizmPrimitiveCalendarModule

Removed module PrizmPrimitiveCalendarRangeModule use standalone PrizmPrimitiveCalendarRangeComponent

Removed deprecated module PrizmPrimitiveMonthPickerModule
Removed module PrizmPrimitiveSpinButtonModule use standalone PrizmPrimitiveSpinButtonComponent

Removed deprecated module PrizmPrimitiveYearMonthPaginationModule
Removed deprecated module PrizmPrimitiveYearPickerModule

Removed module PrizmInputCorrectorModule use standalone PrizmInputCorrectorDirective
Removed module PrizmInputHintModule use standalone PrizmInputHintDirective
Removed module PrizmInputAllowedSymbolsModule use standalone PrizmInputAllowedSymbolsDirective

Removed deprecated module PrizmCardModule
Removed deprecated module PrizmHintModule
Removed deprecated module PrizmScrollControlsModule
Removed deprecated module PrizmInputIconButtonModule
Removed deprecated module PrizmInputDateMultiModule
Removed deprecated module PrizmIndicatorModule
Removed deprecated module PrizmDialogModule
Removed deprecated module PrizmDialogConfirmModule
Removed deprecated module PrizmCronHumanReadableModule
Removed deprecated module PrizmFileUploadModule
Removed deprecated module PrizmCalendarMonthModule
Removed deprecated module PrizmCalendarRangeModule
Removed deprecated module PrizmSidebarModule
Removed deprecated module PrizmColumnSettingsModule

Removed module PrizmErrorPageModule use standalone PrizmErrorPageComponent

### CODE REVIEW

File: apps/doc/src/app/components/counter/counter-example.component.html [hunk 1]
```diff
@@ -78,7 +78,7 @@
       <li>
         <p>
           Import
-          <code>PrizmCardModule</code>
+          <code>PrizmCardComponent</code>
```
Comment (ickisIckis) [inline apps/doc/src/app/components/counter/counter-example.component.html:81]:
```suggestion
          <code>PrizmCounterComponent</code>
```

Comment (ickisIckis) [reply apps/doc/src/app/components/counter/counter-example.component.html:81]: и надо поправить сетап

File: apps/doc/src/app/components/buttons/tree-button/tree-button.module.ts [hunk 2]
```diff
@@ -3,14 +3,14 @@ import { CommonModule } from '@angular/common';
...
 import { TreeButtonComponent } from './tree-button.component';
-import { PrizmTreeButtonModule } from '@prizm-ui/components';
+import { PrizmTreeButtonComponent } from '@prizm-ui/components';
 import { PrizmTreeButtonsExampleComponent } from './examples/tree/tree-button-example.component';
 
 @NgModule({
   imports: [
     CommonModule,
     PrizmAddonDocModule,
-    PrizmTreeButtonModule,
```
Comment (ickisIckis) [inline apps/doc/src/app/components/buttons/tree-button/tree-button.module.ts:13]: у кнопок надо поправить сетап

File: apps/doc/src/app/components/input/textarea/textarea-example.module.ts [hunk 3]
```diff
@@ -6,7 +6,11 @@ import { PrizmAddonDocModule, prizmDocGenerateRoutes } from '@prizm-ui/doc';
 
 import { TextareaBasicExampleComponent } from './examples/textarea-basic-example/textarea-basic-example.component';
 import { TextareaExampleComponent } from './textarea-example.component';
-import { PrizmAutoResizeModule, PrizmInputTextModule, PrizmSelectInputComponent } from '@prizm-ui/components';
+import {
+  PrizmAutoResizeDirective,
+  PrizmInputTextModule,
+  PrizmSelectInputComponent,
```
Comment (ickisIckis) [inline apps/doc/src/app/components/input/textarea/textarea-example.module.ts:12]: а select тут нужен?

File: apps/doc/src/app/components/listing-item/listing-item-example.component.html [hunk 4]
```diff
@@ -90,7 +90,7 @@
       <li>
         <p>
           Import
-          <code>PrizmCardModule</code>
+          <code>PrizmCardComponent</code>
```
Comment (ickisIckis) [inline apps/doc/src/app/components/listing-item/listing-item-example.component.html:93]:
```suggestion
          <code>PrizmListingItemComponent</code>
```

File: apps/doc/src/app/components/skeleton/skeleton.module.ts [hunk 5]
```diff
@@ -6,11 +6,11 @@ import { SkeletonComponent } from './skeleton.component';
...
   PolymorphModule,
   PrizmButtonModule,
-  PrizmCardModule,
+  PrizmCardComponent,
   PrizmInputSelectModule,
   PrizmInputTextModule,
-  PrizmSkeletonModule,
-  PrizmToggleModule,
+  PrizmSkeletonDirective,
+  PrizmToggleComponent,
```
Comment (ickisIckis) [inline apps/doc/src/app/components/skeleton/skeleton.module.ts:13]: надо поправить сетап

File: apps/doc/src/app/components/spinner/spinner.module.ts [hunk 6]
```diff
@@ -4,13 +4,13 @@ import { prizmDocGenerateRoutes, PrizmAddonDocModule } from '@prizm-ui/doc';
...
 import { SpinnerComponent } from './spinner.component';
 import { PrizmSpinnerBaseExampleComponent } from './examples/base/spinner-base-example.component';
-import { PrizmSpinnerModule } from '@prizm-ui/components';
+import { PrizmSpinnerComponent } from '@prizm-ui/components';
 
 @NgModule({
   imports: [
     CommonModule,
     PrizmAddonDocModule,
-    PrizmSpinnerModule,
```
Comment (ickisIckis) [inline apps/doc/src/app/components/spinner/spinner.module.ts:13]: тут тоже, кажется. надо сетап

File: apps/doc/src/app/components/widget/widget-example.module.ts [hunk 7]
```diff
@@ -13,7 +13,7 @@ import { PrizmIconsSvgComponent } from '@prizm-ui/icons';
   imports: [
     CommonModule,
     PrizmAddonDocModule,
-    PrizmWidgetModule,
+    PrizmWidgetComponent,
```
Comment (ickisIckis) [inline apps/doc/src/app/components/widget/widget-example.module.ts:16]: + 1 к сетап

File: libs/components/src/lib/modules/overlay/index.ts [hunk 8]
```diff
@@ -12,5 +13,4 @@ export {
 } from './position';
 export { PrizmOverlayService } from './overlay.service';
 export { PrizmOverlayControl } from './overlay-control';
-export { PrizmOverlayModule } from './overlay.module';
```
Comment (ickisIckis) [inline libs/components/src/lib/modules/overlay/index.ts:15]: не хватает в описании, что его удалили

File: apps/doc/src/app/tools/overlay/overlay.module.ts [hunk 9]
```diff
@@ -17,7 +17,7 @@ import { PrizmOverlayCustomContextExampleComponent } from './examples/custom-con
     CommonModule,
     PrizmAddonDocModule,
     PrizmButtonModule,
-    PrizmOverlayModule,
+    PrizmOverlayComponent,
```
Comment (ickisIckis) [inline apps/doc/src/app/tools/overlay/overlay.module.ts:20]: нужно еще сетап

File: apps/doc/src/app/tools/zone-event/zone-event.module.ts [hunk 10]
```diff
@@ -5,9 +5,9 @@ import { RouterModule } from '@angular/router';
 import { ZoneEventComponent } from './zone-event.component';
 import {
   PrizmButtonModule,
-  PrizmToggleModule,
-  PrizmWidgetModule,
-  PrizmZoneEventModule,
+  PrizmToggleComponent,
```
Comment (ickisIckis) [inline apps/doc/src/app/tools/zone-event/zone-event.module.ts:8]: setup

File: libs/charts/base/src/lib/components/area/index.ts [hunk 11]
```diff
@@ -1,3 +1,2 @@
-export * from './prizm-charts-area.module';
```
Comment (ickisIckis) [inline libs/charts/base/src/lib/components/area/index.ts:1]: пропустили deprecation на все чарты

File: libs/icons/base/src/lib/icons-svg/prizm-icons-svg.module.ts [hunk 12]
```diff
@@ -1,12 +0,0 @@
-import { NgModule } from '@angular/core';
-import { PrizmIconsSvgComponent } from './prizm-icons-svg.component';
-
-/*
- * @deprecated from 4
- * use PrizmIconsComponent instead of this
```
Comment (ickisIckis) [inline libs/icons/base/src/lib/icons-svg/prizm-icons-svg.module.ts:6]: можно обновить аннотацию @deprecated в самом компоненте - там она не корректно выглядит

File: apps/doc/src/app/charts/area/area.module.ts [hunk 13]
```diff
@@ -4,13 +4,13 @@ import { PrizmAddonDocModule, prizmDocGenerateRoutes } from '@prizm-ui/doc';
...
 import { PrizmChartsAreaExampleComponent } from './examples/base/prizm-charts-area-example.component';
-import { PrizmChartsAreaModule } from '@prizm-ui/charts';
+import { PrizmChartsAreaComponent } from '@prizm-ui/charts';
 
 @NgModule({
   imports: [
     CommonModule,
     PrizmAddonDocModule,
-    PrizmChartsAreaModule,
+    PrizmChartsAreaComponent,
```
[the remark this answers is not in the document]
Comment (ZurabDev) [reply apps/doc/src/app/charts/area/area.module.ts:13]: Готово

File: apps/doc/src/app/components/counter/counter-example.component.html (hunk 1 above)
Comment (ZurabDev) [reply apps/doc/src/app/components/counter/counter-example.component.html:81]: Он был поправлен уже

File: apps/doc/src/app/components/buttons/tree-button/tree-button.module.ts (hunk 2 above)
Comment (ZurabDev) [reply apps/doc/src/app/components/buttons/tree-button/tree-button.module.ts:13]: Done

File: apps/doc/src/app/components/input/textarea/textarea-example.module.ts (hunk 3 above)
Comment (ZurabDev) [reply apps/doc/src/app/components/input/textarea/textarea-example.module.ts:12]: Done

File: libs/components/src/lib/modules/overlay/index.ts (hunk 8 above)
Comment (ZurabDev) [reply libs/components/src/lib/modules/overlay/index.ts:15]: Done

File: apps/doc/src/app/tools/zone-event/zone-event.module.ts (hunk 10 above)
Comment (ZurabDev) [reply apps/doc/src/app/tools/zone-event/zone-event.module.ts:8]: Done

File: libs/charts/base/src/lib/components/area/index.ts (hunk 11 above)
Comment (ZurabDev) [reply libs/charts/base/src/lib/components/area/index.ts:1]: Об этом напишем в RFC, их не надо след 2 года хранить

File: libs/icons/base/src/lib/icons-svg/prizm-icons-svg.module.ts (hunk 12 above)
Comment (ZurabDev) [reply libs/icons/base/src/lib/icons-svg/prizm-icons-svg.module.ts:6]: Done

File: apps/doc/src/app/charts/area/area.module.ts (hunk 13 above)
[the remark this answers is not in the document]
Comment (ickisIckis) [reply apps/doc/src/app/charts/area/area.module.ts:13]: возможно, не залилось - не обновлен md все еще + bar, column, gauge, line

File: apps/doc/src/app/components/counter/counter-example.component.html (hunk 1 above)
Comment (ickisIckis) [reply apps/doc/src/app/components/counter/counter-example.component.html:81]: Для card да, тут counter
опечатка в тексте еще - в suggestion

File: apps/doc/src/app/components/buttons/tree-button/tree-button.module.ts (hunk 2 above)
Comment (ickisIckis) [reply apps/doc/src/app/components/buttons/tree-button/tree-button.module.ts:13]: видимо, не запушилось и ниже тоже

Comment (ZurabDev) [reply apps/doc/src/app/components/buttons/tree-button/tree-button.module.ts:13]: Запушил сейчас

File: apps/doc/src/app/components/counter/counter-example.component.html (hunk 1 above)
Comment (ZurabDev) [reply apps/doc/src/app/components/counter/counter-example.component.html:81]: Добавил

File: apps/doc/src/app/charts/area/area.module.ts (hunk 13 above)
[the remark this answers is not in the document]
Comment (ZurabDev) [reply apps/doc/src/app/charts/area/area.module.ts:13]: Сейчас должно быть ок

[the remark this answers is not in the document]
Comment (ickisIckis) [reply apps/doc/src/app/charts/area/area.module.ts:13]: bar, column, gauge, line вот эти еще

File: apps/doc/src/app/components/buttons/tree-button/tree-button.module.ts (hunk 2 above)
Comment (ickisIckis) [reply apps/doc/src/app/components/buttons/tree-button/tree-button.module.ts:13]: еще PrizmButtonModule остался в трех местах (обычные, сплит и иконочные)

File: apps/doc/src/app/components/counter/counter-example.component.html (hunk 1 above)
Comment (ickisIckis) [reply apps/doc/src/app/components/counter/counter-example.component.html:81]: исправим опечатку?
кстати, counter module можно удалить, кажется - там директива и компонент независимо используются

File: apps/doc/src/app/tools/zone-event/zone-event.module.ts (hunk 10 above)
Comment (ickisIckis) [reply apps/doc/src/app/tools/zone-event/zone-event.module.ts:8]: в md не вижу исправления

File: libs/charts/base/src/lib/components/area/index.ts (hunk 11 above)
Comment (ickisIckis) [reply libs/charts/base/src/lib/components/area/index.ts:1]: в discussions добавить

File: apps/doc/src/app/charts/area/area.module.ts (hunk 13 above)
[the remark this answers is not in the document]
Comment (ZurabDev) [reply apps/doc/src/app/charts/area/area.module.ts:13]: Done

File: apps/doc/src/app/components/counter/counter-example.component.html (hunk 1 above)
Comment (ZurabDev) [reply apps/doc/src/app/components/counter/counter-example.component.html:81]: PrizmCounterModule оставим пока для пакетного импорта

File: apps/doc/src/app/components/buttons/tree-button/tree-button.module.ts (hunk 2 above)
Comment (ZurabDev) [reply apps/doc/src/app/components/buttons/tree-button/tree-button.module.ts:13]: Done

File: apps/doc/src/app/tools/zone-event/zone-event.module.ts (hunk 10 above)
Comment (ZurabDev) [reply apps/doc/src/app/tools/zone-event/zone-event.module.ts:8]: Done

File: libs/components/src/lib/components/button/button.module.ts [hunk 14]
```diff
@@ -9,4 +9,4 @@ import { PrizmSplitButtonComponent } from './split-button/split-button.component
   imports: [PrizmButtonComponent, PrizmSplitButtonComponent],
   exports: [PrizmButtonComponent, PrizmSplitButtonComponent],
 })
-export class PrizmButtonModule {}
+export class PrizmButtonComponent {}
```
Comment (ickisIckis) [inline libs/components/src/lib/components/button/button.module.ts:12]: тут, кажется. надо удалить файл и убрать из index.ts

File: libs/components/src/lib/components/counter/counter.module.ts [hunk 15]
```diff
@@ -3,7 +3,6 @@ import { PrizmCounterDirective } from './counter.directive';
 import { PrizmCounterComponent } from './counter.component';
 
 /**
- * @deprecated
```
Comment (ickisIckis) [inline libs/components/src/lib/components/counter/counter.module.ts:6]: оставляем модуль? там два варианта использования, либо директива, либо компонент, вместе их не надо.

Comment (ZurabDev) [reply libs/components/src/lib/components/counter/counter.module.ts:6]: Думаю можно
Done

File: libs/components/src/lib/components/button/button.module.ts (hunk 14 above)
Comment (ZurabDev) [reply libs/components/src/lib/components/button/button.module.ts:12]: Done

**Review by ickisIckis — APPROVED** [review]
LGTM

**Comment by ickisIckis** [comment]
@alexhawkins94, на этапе тестирования нужно проверить разделы setup для удаленных модулей. Там должен быть описан верный импорт: компонент, а не модуль и верноеназвание компонента

**Pull request merged**